lib/codegen/meta-python/isa/x86/recipes.py

```diff
@@ -1473,6 +1486,36 @@ def floatccs(iform):
         disp4(destination, func, sink);
         ''')
 
+indirect_jmp = TailRecipe(
+        'indirect_jmp', BranchTable, size=1, ins=GPR, outs=(),
+        branch_range=32,
```

Comment:
When I try to remove the `branch_range` I get the following:
```
Traceback (most recent call last):
...
  File "/Users/tyler/Code/cranelift/lib/codegen/meta-python/cdsl/isa.py", line 475, in __init__
    .format(recipe, self.inst.name))
AssertionError: Recipe Op1indirect_jmp for indirect_jump_table_br must have a branch_range
```
I poked around for a bit, trying to see if there was some way to change that code in `isa.py` to automatically determine whether the branch needs a range. There doesn't seem to be a clean way to do it.

So the options as I see it:
1. Add a new flag to `Instruction` that says "sure, I'm a branch, but I don't need a range".
2. Remove the check in `isa.py`, and try to avoid shooting ourselves in the foot.
3. Something I haven't thought of.

Any ideas?

Reply:
Hmm, yeah. I don't have any other ideas here. So how about option 1? We can call it `is_indirect_branch`. So it'll look a little redundant `is_branch=True, is_indirect_branch=True`, but fortunately there aren't that many of those :-).